src/index.d.ts

```diff
@@ -11,19 +11,19 @@ type Queries = { [k: string]: QueryReturn<any, mixed> };
 
 type QueriesProps<Decl extends Queries> = FlattenObject<{ [k in keyof Decl]: Decl[k]['_A'] }>;
 
-type QueriesInnerProps<Decl extends Queries> = {
-  [k in keyof Decl]: (
-    | {
-        ready: false;
-      }
-    | {
-        ready: true;
-        value: Decl[k]['_P'];
-      }) & {
-    loading: boolean;
-  }
+type QueryInnerProp<DeclKey> = (
+  | {
+      ready: false;
+    }
+  | {
+      ready: true;
+      value: DeclKey;
+    }) & {
+  loading: boolean;
```

Comment:
@fes300 this typedef seems to be a bit off / outdated: shouldn't it just be `QueryInnerProp = RemoteData<Error, Decl[k]['_P']>`?

Also, what should `Error` be in practice? Without changing how we write queries just yet, this could be set to `Error` or `unknown` I think, nothing more specific

Reply:
at the moment `Error` is whatever value the promise is rejected with, in my idea it coulb be anything as it is defined by the user